spec/requests/tags_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
+
+RSpec.describe 'Tag API', type: :request do
+
+  let!(:tags) {create_list(:tag, 10)}
+  let!(:tag) {tags.first}
+  let(:tag_id) {tags.first.id}
+
+  describe 'GET /tags' do
+    context 'when query parameters are NOT provided' do
+      it 'returns all tags' do
+        get "/tags"
+        json = JSON.parse(response.body)
+        expect(json['tags'].size).to eq(10)
+      end
+    end
+
+    context 'when query parameters are provided' do
+      it 'returns paginated list of tags' do
+        page = 1
+        items_per_page = 10
+        get "/tags?page=#{page}&items_per_page=#{items_per_page}"
+        json = JSON.parse(response.body)
+        expect(json['tags'].size).to eq(10)
+      end
+    end
+  end
+
+  describe 'GET /tags/:id' do
+    it 'gets a tag' do
+      get "/tags/#{tag_id}"
+      json = JSON.parse(response.body)
+      puts "json = #{json}"
+      expect(response).to match_response_schema("tag")
+    end
+  end
+
+  describe 'GET /tags/:name/todos' do
+    it 'gets todos associated with tag name' do
+      todo_1 = tag.todos.create({title: "Todo Item 1"})
```

Comment:
when write long test, there is a good practice to separate test into 3 parts:
1. preparation, such as create data
2. perform action, such as send request, or invoke method
3. assert result

there is a empty line between each parts, and using this practice, I think it is good for other people read and understand your long test :)